include/eve/module/real/core/function/regular/generic/if_else.hpp

```diff
@@ -53,11 +55,26 @@ namespace eve::detail
       using e_t = element_type_t<common_compatible_t<U, V>>;
       using r_t = as_wide_t<e_t, cardinal_t<T>>;
 
-            if constexpr(has_emulated_abi_v<T>)    return map(if_else, cond, r_t(t), r_t(f));
-      else  if constexpr(has_aggregated_abi_v<T>)  return aggregate(if_else, cond, r_t(t), r_t(f));
+            if constexpr(has_emulated_abi_v<T>  ) return map(if_else, cond, r_t(t), r_t(f));
+      else  if constexpr(has_aggregated_abi_v<T>) return aggregate(if_else, cond, r_t(t), r_t(f));
       else  if constexpr(std::same_as<logical<e_t>,element_type_t<T>>)
       {
-        if constexpr( std::same_as<U,V> ) return  bit_select(cond.mask(),r_t(t), r_t(f));
+        if constexpr( std::same_as<U,V> )
+        {
+          if constexpr(kumi::product_type<U> || kumi::product_type<V>)
+          {
+            if constexpr( detail::tag_dispatchable<tag::if_else_,T,U,V> )
```

Comment:
why? What could possible be a reason to customise `if_else`?

Reply:
you have a custom data type from which some member has no business being selected via a mask cause it is only some internal buffers. It cost nothing to do this check anyway so we can have it.